### ISSUE TITLE
Download fails

### ISSUE THREAD

**Issue by CodeYan01** [KRTirtho/spotube#658]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I can play this song https://open.spotify.com/track/0JxsibuxzDijNdlWTlMQ3v from album https://open.spotify.com/album/6ZkiAyyK8QnvWwIfTdhndY but when download it, i get this error

```------- DEVICE INFO -------
computerName: lenovo
numberOfCores: 4
systemMemoryInMegabytes: 16384
------- APP INFO -------
environment: release
version: 3.1.0
appName: spotube
buildNumber: 
packageName: spotube
---------- ERROR ----------
DioException [bad response]: The request returned an invalid status code of 502.
------- STACK TRACE -------
#0      DioMixin.fetch.<anonymous closure> (package:dio/src/dio_mixin.dart:507)
#1      _RootZone.runUnary (dart:async/zone.dart:1661)
#2      _FutureListener.handleError (dart:async/future_impl.dart:165)
#3      Future._propagateToListeners.handleError (dart:async/future_impl.dart:796)
#4      Future._propagateToListeners (dart:async/future_impl.dart:817)
#5      Future._completeError (dart:async/future_impl.dart:592)
#6      _SyncCompleter._completeError (dart:async/future_impl.dart:51)
#7      _Completer.completeError (dart:async/future_impl.dart:23)
#8      Future.any.onError (dart:async/future.dart:620)
#9      _RootZone.runBinary (dart:async/zone.dart:1666)
#10     _FutureListener.handleError (dart:async/future_impl.dart:162)
#11     Future._propagateToListeners.handleError (dart:async/future_impl.dart:796)
#12     Future._propagateToListeners (dart:async/future_impl.dart:817)
#13     Future._completeError (dart:async/future_impl.dart:592)
#14     Future._chainForeignFuture.<anonymous closure> (dart:async/future_impl.dart:537)
#15     _microtaskLoop (dart:async/schedule_microtask.dart:40)
#16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49)
------- CUSTOM INFO -------
```

sometimes status code is 403 like here:
```------- DEVICE INFO -------
computerName: lenovo
numberOfCores: 4
systemMemoryInMegabytes: 16384
------- APP INFO -------
environment: release
version: 3.1.0
appName: spotube
buildNumber: 
packageName: spotube
---------- ERROR ----------
DioException [bad response]: The request returned an invalid status code of 403.
------- STACK TRACE -------
#0      DioMixin.fetch.<anonymous closure> (package:dio/src/dio_mixin.dart:507)
#1      _RootZone.runUnary (dart:async/zone.dart:1661)
#2      _FutureListener.handleError (dart:async/future_impl.dart:165)
#3      Future._propagateToListeners.handleError (dart:async/future_impl.dart:796)
#4      Future._propagateToListeners (dart:async/future_impl.dart:817)
#5      Future._completeError (dart:async/future_impl.dart:592)
#6      _SyncCompleter._completeError (dart:async/future_impl.dart:51)
#7      _Completer.completeError (dart:async/future_impl.dart:23)
#8      Future.any.onError (dart:async/future.dart:620)
#9      _RootZone.runBinary (dart:async/zone.dart:1666)
#10     _FutureListener.handleError (dart:async/future_impl.dart:162)
#11     Future._propagateToListeners.handleError (dart:async/future_impl.dart:796)
#12     Future._propagateToListeners (dart:async/future_impl.dart:817)
#13     Future._completeError (dart:async/future_impl.dart:592)
#14     Future._chainForeignFuture.<anonymous closure> (dart:async/future_impl.dart:537)
#15     _microtaskLoop (dart:async/schedule_microtask.dart:40)
#16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49)
------- CUSTOM INFO -------
```

This happens on both Win 10 and android

I find many songs in https://open.spotify.com/album/6ZkiAyyK8QnvWwIfTdhndY and https://open.spotify.com/album/2LYWVRbZGHUuXeMqVmtt2w where download doesn't work but i can play them just fine. 

### Expected Behavior

If I download the track, it should download.

### Steps to reproduce

1. Opened the app.
2. Went to Library > Albums
3. Opened the album https://open.spotify.com/album/2LYWVRbZGHUuXeMqVmtt2
4. Selected the song and downloaded it.
5. Error downloading. Try redownloading and still won't work.

### Operating System

Windows and Android

### Spotube version

3.1.0

### Installation source

Play Store (Android)

### Additional information

On windows i installed from the website.

Many songs don't download from those albums, and those are just the first 2 albums i'm trying on the app.

I see https://github.com/KRTirtho/spotube/issues/644 which looks similar, but the error message is different and is due to unsanitized filename. Might be another case of unsanitized, but i wanted to report it as it may be in a different part of the code.

**Comment by KRTirtho** [comment]
It has been fixed in the [nightly release](https://spotube.krtirtho.dev/other-downloads/nightly-downloads)

For stable version (<=3.1.0) there's a workaround available:
- Change "Settings" > "API Type" to "Piped" temporarily
- Change "Settings" > "Piped Instance" to "drgns.space" (this is the only instance working)
- Change "Settings" > "API Type" back to "Youtube"
- Cancel all download, restart the app & finally re-download those tracks

hope it helps

**Comment by CodeYan01** [comment]
workaround didn't work for me on windows, but can confirm nightly release works. thanks!

**Comment by CodeYan01** [comment]
So nightly release did work, but apparently as of this writing, the file name sanitization isn't part of the nightly build yet, so it still fails. it's fine though, i can wait for the next nightly

**Comment by KRTirtho** [comment]
Fixed in v3.1.1
